### PR TITLE
Add comm=ugni perfstats to measure dynamic alloc related activities.

### LINKED ISSUE #8625: Investigate overheads of dynamic array registration
https://github.com/chapel-lang/chapel/pull/6947 added dynamic array registration support for ugni in order to improve numa affinity for arrays. We've had some outstanding questions about the overheads and non-numa-affinity performance implications of this work, particularly with regard to ISx.

We thought the broadcast critical section might be slow and that it could have a significant serializing impact on applications. We took a look reducing the critical section in https://github.com/chapel-lang/chapel/pull/8177, but we didn't see any large performance wins with that. Prior to https://github.com/chapel-lang/chapel/issues/8497 we also didn't have a good platform to investigate because ISx was so noisy. 

This issue is meant to serve as a placeholder to investigate as well as a place to store the results for future reference.

### PR DESCRIPTION
**Pull request by gbtitus**
This adds some more internal performance statistics, this time having to
do with dynamic allocation.  I also did some cleaning up in perfstats,
reducing the number of times we repeat the counter type and things like
that.  All of this is normally turned off; we only use it for low-level
performance investigations because it does slow down normal operations
noticeably.

Measurements produced by these changes led to the conclusions which
allowed us to close #8625.

### CODE REVIEW

File: runtime/src/comm/ugni/comm-ugni.c [hunk 1]
```diff
@@ -315,28 +331,52 @@ chpl_comm_pstats_t chpl_comm_pstats;
...
+#define PERFSTATS_TSTAMP(ts)    _PSV_C_TYPE ts = perfstats_timer_get()
+#define PERFSTATS_TDIFF(te, ts) ((te) - (ts) - timer_overhead)
+#define PERFSTATS_TELAPSED(ts)  PERFSTATS_TDIFF(perfstats_timer_get(), ts)
 
 static void perfstats_init(void)
 {
   (void) clock_gettime(CLOCK_MONOTONIC, &perfstats_timeBase);
+
+  //
+  // Measure the timer overhead.  We only use timer_nsecs so that we
```
Comment (ronawho) [inline runtime/src/comm/ugni/comm-ugni.c]: If you're worried about timer overhead you could consider using `CLOCK_MONOTONIC_COARSE` instead (requires 2.6.32, but that shouldn't be an issue for ugni)

Comment (ronawho) [reply runtime/src/comm/ugni/comm-ugni.c]: Although it's less accurate, so maybe that's not what you want anyways

Comment (gbtitus) [reply runtime/src/comm/ugni/comm-ugni.c]: Yeah, these are fairly fine-grained measurements in some cases.

File: runtime/src/comm/ugni/comm-ugni.c [hunk 2]
```diff
@@ -2975,6 +3048,17 @@ void chpl_comm_impl_regMemPostAlloc(void* p, size_t size)
   mem_region_t* mr;
   int mr_i;
 
+#if 0
```
Comment (ronawho) [inline runtime/src/comm/ugni/comm-ugni.c]: I don't love the `if 0` uses, I would prefer we just use a new define to trigger this. i.e. `#if PERFSTATS_COMM_UGNI_ZERO_ARRAY_ALLOC` or something

**Pull request merged**